R/plot_plate.R

```diff
@@ -0,0 +1,273 @@
+require(ggplot2)
+require(grid)
+require(png)
+
+
+#' Plot a 96-well plate with colored wells
+#'
+#' It is a generic function to plot a 96-well plate with colored wells
+#' used by other functions in this package, mainly to plot layout and counts.
+#' The function uses a background image of a 96-well plate and
+#' plots the colors in the wells using ggplot2.
+#'
+#' @param colors A vector with 96 colors
+#' @param plot_numbers Logical indicating if the well numbers should be plotted
+#' @param numbers A vector with 96 numbers
+#' @param plot_title The title of the plot (default is "Plate")
+#' @param plot_legend Logical indicating if the legend should be plotted
+#' @param legend_mapping A named vector with the colors mapping
+#'
+#' @return A ggplot object
+#'
+#' @import ggplot2
+#' @import grid
+#' @import png
+#'
+#' @internal
+plot_plate <- function(colors, plot_numbers = FALSE, numbers = NULL, plot_title = "Plate",
+                       plot_legend = FALSE, legend_mapping = NULL) {
```

Comment:
We can remove the need for `plot_numbers` and `plot_legend` flags and the just use the value from the legend and numbers. Ex. If the numbers vector is provided then plot the number is it is NULL then don't

Reply:
`legend_mapping` is always passed to create plots, I could change this in the case of `plot_legend` but then we would have two "ways" of handling this and on top of that I think current approach is more direct

Reply:
Okey